demos/gaze_estimation_demo/cpp/src/landmarks_estimator.cpp

```diff
@@ -99,10 +98,7 @@ std::vector<cv::Point2i> LandmarksEstimator::heatMapPostprocess(cv::Rect faceBou
 
 std::vector<cv::Mat> LandmarksEstimator::split(std::vector<float>& data, const std::vector<unsigned long>& shape) {
     std::vector<cv::Mat> flattenData(shape[1]);
-    float* output = new float[data.size()];
-    for (size_t i = 0; i < data.size(); i++) {
-        output[i] = data[i];
-    }
+    float* output = &data.front();
```

Comment:
```suggestion
    float* output = data.data;
```
But I'm not sure `data` vector outlives `flattenData`. Is this guaranteed?

Reply:
Even better. `flattenData[i] = cv::Mat(shape[2], shape[3], CV_32FC1, output + i * shape[2] * shape[3]);` can be replaced with `flattenData[i] = cv::Mat(shape[2], shape[3], CV_32FC1, data.data + i * shape[2] * shape[3]);`

Reply:
Fixed.
But function returns copy of vector `flattenData`, isn't it? Is this full copy of `std::vector<cv::Mat>`? Now landmarks work correctly.

Reply:
It copies mats, not underlying data:
> [Matrix constructors that take data and step parameters do not allocate matrix data.](https://docs.opencv.org/4.x/d3/d63/classcv_1_1Mat.html#a51615ebf17a64c968df0bf49b4de6a3a)

Reply:
Ok. Memory isn't shared between threads and lives as long as the result (`std::vector<cv::Point2i>`) is built with this memory. I don't know the meaning of this `new`.

Reply:
I'm not sure I follow. To me the hypothetical problem is that mats stored in `flattenData` borrow memory from `data` vector. This can be a real problem if `flattenData` outlives `data`.

Reply:
Revert.